chainer/exporters/caffe.py

```diff
@@ -0,0 +1,458 @@
+                f = open(self.prototxt, 'wt')
+                f.write('name: "{}"\n'.format(name))
+                assert len(inputs) == 1
+                f.write('layer {\n'
+                        '  name: "data"\n'
+                        '  type: "Input"\n'
+                        '  top: "data"\n'
+                        '  input_param { shape: {')
+                for i in inputs[0].shape:
+                    f.write(' dim: ' + str(i))
+                f.write(' } }\n'
+                        '} \n')
+            for i in dumped_list:
+                self.dump_function_object(i, f, net)
+        finally:
+            if f is not None:
+                f.close()
+
+        if net is not None:
+            with open(self.caffemodel, 'wb') as f:
+                f.write(net.SerializeToString())
+            if self.debug:
+                import google.protobuf.text_format
+                with open(self.caffemodel + ".txt", 'w') as f:
+                    f.write(google.protobuf.text_format.MessageToString(net))
+
+
+def export(model, args, directory=None,
+           export_params=True, graph_name='Graph'):
+    """(Experimental) Export a computational graph as Caffe format.
+
+    Args:
+        model (~chainer.Chain): The model object you want to export in ONNX
+            format. It should have :meth:`__call__` method because the second
+            argment ``args`` is directly given to the model by the ``()``
+            accessor.
+        args (list of ~chainer.Variable): The argments which are given to the
+            model directly.
+        directory (str): The directory used for saving the resulting Caffe
+            model. If None, nothing is saved to the disk.
+        export_params (bool): If True, this function exports all the parameters
+            included in the given model at the same time. If False, the
+            exported Caffe model doesn't include any parameter values.
+        graph_name (str): A string to be used for the ``name`` field of the
+            graph in the exported Caffe model.
+
+    .. note::
+        Currently, this function supports networks that created by following
+        layer functions.
+
+        - :func:`~chainer.functions.linear`
+        - :func:`~chainer.functions.convolution_2d`
+        - :func:`~chainer.functions.deconvolution_2d`
+        - :func:`~chainer.functions.max_pooling_2d`
+        - :func:`~chainer.functions.average_pooling_2d`
+        - :func:`~chainer.functions.batch_normalization`
+        - :func:`~chainer.functions.local_response_normalization`
+        - :func:`~chainer.functions.relu`
+        - :func:`~chainer.functions.concat`
+        - :func:`~chainer.functions.softmax`
+        - :func:`~chainer.functions.reshape`
+        - :func:`~chainer.functions.add`
+
+        This function can export at least following networks.
+
+        - GoogLeNet
+        - ResNet
+        - VGG
+
+        And, this function use testing (evaluation) mode.
```

Comment:
It might not be enough for users. Is there any intention to omit the explanation about it forces backprop mode?

Reply:
I think using force backprop mode is not important information for users.

Reply:
I see. Now I think this docstring looks fine.